File: Go/019. Remove Nth Node From End of List.go
type ListNode struct {
    Val int
    Next *ListNode
}

func removeNthFromEnd(head *ListNode, n int) *ListNode {
	res := &ListNode{0, head}
	front, rear := res, res
	var i int
	for i = 0; i < n && rear.Next != nil; i++ {
		rear = rear.Next
	}
	if i < n {
		return res.Next
	}
	for rear.Next != nil {
		front = front.Next
		rear = rear.Next
	}
	front.Next = front.Next.Next
	return res.Next
}

File: C++/019. Remove Nth Node From End of List.cpp
#include <iostream>
#include <vector>

struct ListNode {
    int val;
    ListNode *next;
    ListNode(int x) : val(x), next(NULL) {}
};

class Solution {
public:
    ListNode* removeNthFromEnd(ListNode* head, int n) {
        ListNode *res = new ListNode(0);
        res->next = head;
        ListNode *front = res, *rear = res;
        int i;
        for (i = 0; i < n && rear->next != nullptr; i++) {
            rear = rear->next;
        }
        if (i < n) {
            return rear->next;
        }
        while (rear->next != nullptr) {
            front = front->next;
            rear = rear->next;
        }
        front->next = front->next->next;
        return res->next;
    }
};